File: JSBASIC/118_smallest_digit.js
// 118	118_smallest_digit.js	Find the smallest digit in a given number.

let num = 39271;
let smallest = 9;
let temp = num;
while (temp > 0) {
    let digit = temp % 10;
    if (digit < smallest) smallest = digit;
    temp = Math.floor(temp / 10);
}
console.log("Number:", num);
console.log("Smallest digit:", smallest);

File: JavaBasic/118_smallest_digit.java
// 118	118_smallest_digit.java	Find the smallest digit in a given number.
public class _118_smallest_digit {
    public static void main(String[] args) {
        int num = 39271, smallest = 9, temp = num;
        while (temp > 0) { int d = temp%10; if(d<smallest) smallest=d; temp/=10; }
        System.out.println("Number: " + num + " | Smallest digit: " + smallest);
    }
}
